app/controllers/katello/api/v2/sync_plans_controller.rb

```diff
@@ -92,6 +92,7 @@ def destroy
     param :id, String, :desc => N_("ID of the sync plan"), :required => true
     param :product_ids, Array, :desc => N_("List of product ids to add to the sync plan"), :required => true
     def add_products
+      @sync_plan.validate_and_update_products force_update: true
```

Comment:
It looks like running an update on the sync plan also causes invalid products to error out, even when changing innocuous fields like the description. Perhaps we should add this check for updates as well.

Reply:
Pushed a commit to validate before update.